src/halmos/solve.py

```diff
@@ -27,6 +27,34 @@
 EXIT_TIMEDOUT = 124
 
 
+@dataclass
+class DumpDirectory:
+    """Wrapper for directory used for dumping SMT files.
+    
+    Can wrap either a TemporaryDirectory or a regular directory path.
+    Provides a consistent .name property for both cases.
+    """
+    _temp_dir: TemporaryDirectory | None = None
+    _path: str | None = None
+    
+    @classmethod
+    def from_temp_dir(cls, temp_dir: TemporaryDirectory) -> "DumpDirectory":
+        return cls(_temp_dir=temp_dir)
+    
+    @classmethod 
+    def from_path(cls, path: str | Path) -> "DumpDirectory":
+        return cls(_path=str(path))
+    
+    @property
+    def name(self) -> str:
+        if self._temp_dir is not None:
+            return self._temp_dir.name
+        elif self._path is not None:
+            return self._path
+        else:
+            raise ValueError("DumpDirectory not properly initialized")
```

Comment:
we don't need to introduce a whole new dataclass for this, instead introduce a simple `DumpDirectory` type alias for `TemporaryDirectory | Path`

then we can just have a helper function `dirname(dump_dir: DumpDirectory)` that does a match on the actual type of dump_dir. Basically like a lightweight algebraic data type.